unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarVisibility.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace DCL
+{
+    public class AvatarVisibility : MonoBehaviour
+    {
+
+        public AvatarRenderer renderer;
+        [Tooltip("Game objects that should be hidden/shown together with the renderer")]
+        public GameObject[] gameObjectsToToggle;
+        // When this property is true, then the visibility can't be modified
+        private bool locked = false;
+        private bool? isVisible;
+
+        public void SetVisibility(bool visibility)
+        {
+            if (locked == false && isVisible != visibility)
+            {
+                if (visibility)
+                {
+                    renderer.ShowAllRenderers();
+                }
+                else
+                {
+                    renderer.HideAllRenderers();
+                }
+                foreach (GameObject gameObject in gameObjectsToToggle)
+                {
+                    gameObject.SetActive(visibility);
+                }
+
+                isVisible = visibility;
+            }
+
+        }
+
+        public void SetLock(bool newLockValue)
+        {
+            locked = newLockValue;
+        }
```

Comment:
We should probably call `if(!locked) SetVisibility(isVisible);` after this lone line in `SetLock()` to resume this class's function once it is unlocked. 

What do you think?